### ISSUE TITLE
Relationship with upstream maintainer/s

### ISSUE THREAD

**Issue by cokiengchiara** [giantswarm/roadmap#339]
**Context**
* We say one value we provide re: managed apps is that we work with upstream and can get questions and issues resolved ('one hand to shake' https://github.com/giantswarm/rfc/pull/16#issuecomment-785955558)
* Currently struggling to get a response from both Kong https://discuss.konghq.com/t/missing-kong-plugin-name-in-logs/8369 and Aqua https://community.grafana.com/t/how-move-chunks-and-index-files-from-s3-to-local-filesystem/40867/7

**Idea: Part of deciding to move an app from 'best effort' to 'managed' should be to start building a relationship w upstream maintainer. To start, introduce ourselves and do an intro call with an upstream maintainer.**

If you agree this is a good idea, I think we should
1. Brainstorm what value we can provide - e.g. report detailed bugs, convey important feature requests, etc. It would be a good idea to commit to providing value on a regular basis.
2. Reach out to traines from Kong to get on a call
3. Reach out to ewelch from Grafana to get on a call

@giantswarm/team-halo cc @Jul85 (sig partner)

**Comment by cokiengchiara** [comment]
**Summary**

1. How to establish initial relationship
2. When someone already established initial relationship, how to connect w others in the team (e.g. make an intro)
3. Have doc in intranet detailing 1 and 2.

**Loki issue on data stuff**

- [ ] @piontec will ping the person on Slack re: Loki for the data migration

**Linkerd**

- [ ] @cokiengchiara can post on their Linkerd.

> Hi, we are from Giant Swarm. We're using X. It would be nice to know who to reach out to when we encounter problems. We work on a lot of helm charts and know them well. Ex -- . So you can expect some inputs from us in this area.

**In general**
- [ ] Create intranet page on who has relationship with whom from upstream, and how to contact


Notes
* Need a go-to person and their preferred mode of communication
* How to deal w upstream maintainer? We don't want to bypass their support plans? 
* Lukasz and Julia already tried to get in contact w Grafana - from both commercial and community perspective. They didn't really care, but mostly marketing perspective.
* What about technical perspective?

**Comment by cokiengchiara** [comment]
> * [x]  @cokiengchiara can post on their Linkerd.
> 
> > Hi, we are from Giant Swarm. We're using X. It would be nice to know who to reach out to when we encounter problems. We work on a lot of helm charts and know them well. Ex -- . So you can expect some inputs from us in this area.

i messaged on their slack: https://share.getcloudapp.com/jkuPdzk8

Also started this 'Halo upstream technical contacts' spreadsheet: https://docs.google.com/spreadsheets/d/10jhzZJNmGA6yrXgsM8Lx5mkhJF0sVZ-Jai8snIRGZx4/edit?usp=sharing

**Comment by cokiengchiara** [comment]
> * [x]  @cokiengchiara can post on their Linkerd.
> 
> > Hi, we are from Giant Swarm. We're using X. It would be nice to know who to reach out to when we encounter problems. We work on a lot of helm charts and know them well. Ex -- . So you can expect some inputs from us in this area.

i messaged on their slack: https://share.getcloudapp.com/jkuPdzk8

Also started this 'Halo upstream technical contacts' spreadsheet: https://docs.google.com/spreadsheets/d/10jhzZJNmGA6yrXgsM8Lx5mkhJF0sVZ-Jai8snIRGZx4/edit?usp=sharing

**Comment by cokiengchiara** [comment]
Tue, 25. May 2021
Chiara w Jason Morgan (Linkerd technical evangelist)

tl;dr Linkerd has a “Linkerd enablement program”. There is no cost or commitment. Their aim is to get the message out about Linkerd and help people who use Linkerd succeed at it.

- For technical, Team Halo, just join Linkerd Slack when you have a question https://slack.linkerd.io/ 
- For business / partnership, Julia, maybe you want to engage with Jason Morgan? 


More details
(1) Technical -- he’s happy to get our technical questions answered in https://slack.linkerd.io/ Whenever we’re ready, they’re happy to do a ‘training’ for our engineers, either a Linkerd 101 or whatever we need help with. After that we can do regular (monthly / quarterly) trainings, they’re happy to show up as well when we do these for our customers.
(2) Business / Marketing -- they can do co-marketing, help us get talks submitted to CNCF if we have an interesting customer story on Linkerd, send ‘consulting’ work our way, etc. I’m waiting for him to send some slides.

Even more info: 

- Due to difficulties getting technical questions answered for Kong and Loki, Team Halo is experimenting with building relationships w upstream earlier on.
- I joined the Linkerd Slack and had a brief call with Jason Morgan. He’s technical evangelist at Buoyant, the company behind Linkerd.
- Buoyant is a small-ish company (23 employees on LinkedIn) - “There are more folks that market Istio than work at Buoyant”.

**Comment by cokiengchiara** [comment]
RE: Linkerd specifically, that was quite easy. I'm happy to try it again on other apps 😂

**Comment by Jul85** [comment]
Hi @cokiengchiara : 

I Just had a really nice call with Catherine Paganini of Linkderd. She was really interestedin doing sth. together with us. Here are my meeting notes:

- good start would be a joint blogpost (GS and together with Jason) - that would live at their buoanyt page and would be distributed from there as well, but we could also put it on our website and use it for our purposes of course
- after the blogpost, we could reuse its content for different other formats together, if we like (e.g.webinar, video, talks at meetups or conferences, but that is really up to us, seeing a benefit in that)
- another possibility would be: they are looking for partner for their buoyant cloud (for managed linkerd) - but we are managing linkerd ourselves?

Next steps: 
- call with Jason and GS to talk about the details of a blogpost.

ping @HelloHeatherC for assistance also.

**Comment by HelloHeatherC** [comment]
Here's the old ticket for reference: https://github.com/giantswarm/giantswarm/issues/15480

**Comment by cokiengchiara** [comment]
Yey!

> * another possibility would be: they are looking for partner for their buoyant cloud (for managed linkerd) - but we are managing linkerd ourselves?

What would it mean to be partner for their buoyant cloud? Wouldn't it be the same for us offering a managed Linkerd and they help promote it because it evangalizes their Linkerd product (not service)?

**Comment by cokiengchiara** [comment]
@cokiengchiara will add it to quality bars. It's a suggestion.

**Comment by cokiengchiara** [comment]
Moving back to backlog. Maybe new PO would like to do sth about it.

**Comment by JosephSalisbury** [comment]
We should be friends with maintainers, yes
